### PR TITLE
feat: twitter actions for account mentions and post tweet reply

### PR DESCRIPTION
**Pull request by stat**

### CODE REVIEW

**Comment by cb-heimdall** [comment]
<h1>🟡 Heimdall Review Status</h1>
<table>
<tr>
<th>Requirement</th>
<th>Status</th>
<th>More Info</th>
</tr>
<tr>
<td>Reviews</td>
<td>
🟡
<code>0/1</code>
</td>
<td>
<details>
<summary>Denominator calculation</summary>
<table>
<tr>
<td>
<details>
<summary>Show calculation</summary>
<table>
<tr>
<td><code>1</code> if user is bot</td>
<td>0</td>
</tr>
<tr>
<td><code>1</code> if user is external</td>
<td>0</td>
</tr>
<tr>
<td>
From <code>.codeflow.yml</code>
</td>
<td>1</td>
</tr>
<tr>
<td>
Additional review requirements
<details>
<summary>Show calculation</summary>
<table>
<tr>
<td><strong>Max</strong></td>
<td><strong>0</strong></td>
</tr>
</table>
</details>
</td>
<td>0</td>
</tr>
<tr>
<td>
From <code>CODEOWNERS</code>
</td>
<td>0</td>
</tr>
<tr>
<td>Global minimum</td>
<td>0</td>
</tr>
<tr>
<td><strong>Max</strong></td>
<td>
<strong>
1
</strong>
</td>
</tr>
</table>
</details>
</td>
<td>1</td>
</tr>
<tr>
<td><code>1</code> if commit is unverified</td>
<td>0</td>
</tr>
<tr>
<td><strong>Sum</strong></td>
<td><strong>1</strong></td>
</tr>
</table>
</details>
</td>
</tr>
</table>

File: twitter-langchain/examples/account_details/account_details.py [hunk 1]
```diff
@@ -44,15 +44,11 @@
...
-#  {
-#      "id": "1234567890123456789",
-#      "name": "My Twitter Name",
-#      "username": "MyTwitterUserName",
-#      "link": "https://x.com/MyTwitterUserName"
-#  }
+#  Here are your Twitter account details:
+
+#  - **Name: ** CDP AgentKit
+#  - **Username: ** [@CDPAgentKit](https: // x.com/CDPAgentKit)
```
Comment (rohan-agarwal-coinbase) [inline twitter-langchain/examples/account_details/account_details.py:53]: are the spaces in the URL intentional?

File: twitter-langchain/examples/account_details/pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+cdp-agentkit-core = {path = "../../../cdp-agentkit-core", develop = true}
+cdp-langchain = {path = "../../../cdp-langchain", develop = true}
+twitter-langchain = {path = "../../../twitter-langchain", develop = true}
+langchain = "^0.3.4"
+langchain-openai = "^0.2.4"
+langgraph = "^0.2.39"
+python = "^3.10"
+tweepy = "^4.14.0"
+
+[tool.poetry.group.dev.dependencies]
```
Comment (rohan-agarwal-coinbase) [inline twitter-langchain/examples/account_details/pyproject.toml:22]: can this be deleted?

File: twitter-langchain/examples/chatbot/.env-local [hunk 3]
```diff
@@ -0,0 +1,6 @@
+	OPENAI_API_KEY=
```
Comment (rohan-agarwal-coinbase) [inline twitter-langchain/examples/chatbot/.env-local:1]: nit: remove the left trailing whitespace in these lines

Comment (rohan-agarwal-coinbase) [inline twitter-langchain/examples/chatbot/.env-local:1]: nit: remove the left trailing whitespace in these lines

File: twitter-langchain/examples/chatbot/README.md [hunk 4]
```diff
@@ -0,0 +1,51 @@
+# CDP Agentkit Twitter Langchain Extension Examples - Chatbot
+
+This example demonstrates an agent setup as a terminal style chatbot with access to Twitter (X) API actions.
+
+## Ask the chatbot to engage in the Twitter (X) ecosystem!
+- "Transfer a portion of your ETH to john2879.base.eth"
```
Comment (rohan-agarwal-coinbase) [inline twitter-langchain/examples/chatbot/README.md:6]: this doesn't work without access to other actions right?

File: twitter-langchain/examples/chatbot/chatbot.py [hunk 5]
```diff
@@ -0,0 +1,132 @@
+import asyncio
```
Comment (rohan-agarwal-coinbase) [inline twitter-langchain/examples/chatbot/chatbot.py:1]: are these unused imports now?

**Review by rohan-agarwal-coinbase — DISMISSED** [review]

File: twitter-langchain/examples/chatbot/README.md (hunk 4 above)
Comment (stat) [reply twitter-langchain/examples/chatbot/README.md:6]: hah right, i left it in as tribute i suppose

File: twitter-langchain/examples/chatbot/.env-local (hunk 3 above)
Comment (stat) [reply twitter-langchain/examples/chatbot/.env-local:1]: roger that

Comment (stat) [reply twitter-langchain/examples/chatbot/.env-local:1]: https://developer.x.com/en/portal/dashboard

File: twitter-langchain/examples/chatbot/chatbot.py (hunk 5 above)
Comment (stat) [reply twitter-langchain/examples/chatbot/chatbot.py:1]: yeah, not sure how these passed the linter

File: twitter-langchain/examples/account_details/account_details.py (hunk 1 above)
Comment (stat) [reply twitter-langchain/examples/account_details/account_details.py]: deleted example

**Review by rohan-agarwal-coinbase — APPROVED** [review]

**Review by rohan-agarwal-coinbase — DISMISSED** [review]

**Comment by cb-heimdall** [comment]
Review Error for rohan-agarwal-coinbase @ 2024-11-15 21:01:11 UTC
User failed mfa authentication, see [go/mfa-help](http://go/mfa-help)

**Review by rohan-agarwal-coinbase — DISMISSED** [review]
CHANGELOG entry

**Review by rohan-agarwal-coinbase — APPROVED** [review]

**Pull request merged**